sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/HttpInstrumentationOptions.java

```diff
@@ -0,0 +1,299 @@
+        return isHttpLoggingEnabled;
+    }
+
+    /**
+     * Flag indicating whether HTTP request and response header values are added to the logs
+     * when their name is not explicitly allowed via {@link HttpInstrumentationOptions#setAllowedHeaderNames(Set)} or
+     * {@link HttpInstrumentationOptions#addAllowedHeaderName(HttpHeaderName)}.
+     * True by default.
+     *
+     * @return True if redacted header names logging is enabled, false otherwise.
+     */
+    public boolean isRedactedHeaderNamesLoggingEnabled() {
+        return isRedactedHeaderNamesLoggingEnabled;
+    }
+
+    /**
+     * Enables or disables logging of redacted header names.
+     * @param redactedHeaderNamesLoggingEnabled True to enable logging of redacted header names, false otherwise.
+     *                                          Default is true.
+     * @return The updated {@link HttpInstrumentationOptions} object.
+     */
+    public HttpInstrumentationOptions setRedactedHeaderNamesLoggingEnabled(boolean redactedHeaderNamesLoggingEnabled) {
+        isRedactedHeaderNamesLoggingEnabled = redactedHeaderNamesLoggingEnabled;
+        return this;
+    }
+
+    /**
+     * Flag indicating whether HTTP request and response body is logged.
+     * False by default.
+     * <p>
+     * Note: even when content logging is explicitly enabled, content is not logged
+     * for requests and responses where the content length is not known or greater than 16KB.
+     *
+     * @return True if content logging is enabled, false otherwise.
+     */
+    public boolean isContentLoggingEnabled() {
+        return isContentLoggingEnabled;
+    }
+
+    /**
+     * Enables or disables logging of HTTP request and response.
+     * False by default.
+     * <p>
+     * When HTTP logging is disabled, basic information about the request and response is still recorded
+     * via distributed tracing.
+     *
+     * @param isHttpLoggingEnabled True to enable detailed HTTP logging, false otherwise.
+     * @return The updated {@link HttpInstrumentationOptions} object.
+     */
+    public HttpInstrumentationOptions setHttpLoggingEnabled(boolean isHttpLoggingEnabled) {
+        this.isHttpLoggingEnabled = isHttpLoggingEnabled;
+        return this;
+    }
+
+    /**
+     * Enables or disables logging of HTTP request and response body. False by default.
+     * Enabling content logging also enables HTTP logging in general.
+     * <p>
+     * Note: even when content logging is explicitly enabled, content is not logged for requests and responses where the
+     * content length is not known or greater than 16KB.
+     *
+     * @param isContentLoggingEnabled True to enable content logging, false otherwise.
+     * @return The updated {@link HttpInstrumentationOptions} object.
+     */
+    public HttpInstrumentationOptions setContentLoggingEnabled(boolean isContentLoggingEnabled) {
+        this.isHttpLoggingEnabled |= isContentLoggingEnabled;
```

Comment:
I feel with this existing we may want to change `setHttpLoggingEnabled` back to being an enum that has greater flexibility and more verbosity.

Reply:
I brought `HttpLogLevelDetail` back.

I'd prefer 

```java
new HttpInstrumentationOptions().setHttpLogsEnabled(true)
```

over

```java
new HttpInstrumentationOptions()
     .setHttpLogLevel(HttpInstrumentationOptions.HttpLogDetailLevel.HEADERS))
```

for the most popular use-case, but given the latter is more extensible, let's do it. I'll probably bikeshed on this more to make it shorter